lib/spack/llnl/util/cpu/detect.py

```diff
@@ -109,20 +109,64 @@ def sysctl(*args):
         'model name': sysctl('-n', 'machdep.cpu.brand_string')
     }
 
-    # Super hacky way to deal with slight representation differences
-    # Would be better to somehow consider these "identical"
-    if 'sse4.1' in info['flags']:
+    adjust_raw_flags(info)
+
+    return info
+
+
+def adjust_raw_flags(info):
+    """Adjust the flags detected on the system to homogenize
+    slightly different representations.
+    """
+    # Flags detected on Darwin turned to their linux counterpart
+    flags = info.get('flags', [])
+    if 'sse4.1' in flags:
         info['flags'] += ' sse4_1'
-    if 'sse4.2' in info['flags']:
+    if 'sse4.2' in flags:
         info['flags'] += ' sse4_2'
-    if 'avx1.0' in info['flags']:
+    if 'avx1.0' in flags:
         info['flags'] += ' avx'
-    if 'clfsopt' in info['flags']:
+    if 'clfsopt' in flags:
         info['flags'] += ' clflushopt'
-    if 'xsave' in info['flags']:
+    if 'xsave' in flags:
         info['flags'] += ' xsavec xsaveopt'
 
-    return info
+
+def adjust_raw_vendor(info):
+    """Adjust the vendor field to make it human readable"""
+    # Mapping numeric codes to vendor (ARM). This list is a merge from
+    # different sources:
+    #
+    # https://github.com/karelzak/util-linux/blob/master/sys-utils/lscpu-arm.c
+    # https://developer.arm.com/docs/ddi0487/latest/arm-architecture-reference-manual-armv8-for-armv8-a-architecture-profile
+    # https://github.com/gcc-mirror/gcc/blob/master/gcc/config/aarch64/aarch64-cores.def
+    # https://patchwork.kernel.org/patch/10524949/
+    arm_vendor = {
+        '0x41': 'ARM',
+        '0x42': 'Broadcom',
+        '0x43': 'Cavium',
+        '0x44': 'DEC',
+        '0x46': 'Fujitsu',
+        '0x48': 'HiSilicon',
+        '0x49': 'Infineon Technologies AG',
+        '0x4d': 'Motorola',
+        '0x4e': 'Nvidia',
+        '0x50': 'APM',
+        '0x51': 'Qualcomm',
+        '0x53': 'Samsung',
+        '0x56': 'Marvell',
+        '0x61': 'Apple',
+        '0x66': 'Faraday',
+        '0x68': 'HXT',
+        '0x69': 'Intel'
+    }
```

Comment:
I think the mapping above should go in the JSON file, as it is essential for interpreting the vendors there.  If this lives in Python, implementors will have to duplicate the information if they make `archspec` bindings for other languages.

Reply:
Done